src/PCRE2Engine.h

```diff
@@ -0,0 +1,28 @@
+// -*- c++ -*-
+#ifndef REGEXBENCH_PCRE2_H
+#define REGEXBENCH_PCRE2_H
+
+#define PCRE2_CODE_UNIT_WIDTH 8
+#include <pcre2.h>
+
+#include "Engine.h"
+
+namespace regexbench {
+
+class PCRE2Engine : public Engine {
+public:
+  PCRE2Engine();
+  virtual ~PCRE2Engine();
+
+  virtual void compile(const std::vector<Rule> &);
+  virtual bool match(const char *, size_t);
+
+private:
+  std::vector<pcre2_code *>res;
```

Comment:
Make this `vector<unique_ptr<pcre2_code, decltype(pcre2_code_free)>>` so that we don't have to call `pcre2_code_free()` explicitly in the destructor.